cpp/src/arrow/flight/sql/odbc/odbcabstraction/calendar_utils.cc

```diff
@@ -39,13 +41,30 @@ int64_t GetTodayTimeFromEpoch() {
 #endif
 }
 
-void GetTimeForSecondsSinceEpoch(tm& date, int64_t value) {
-#if defined(_WIN32)
-  gmtime_s(&date, &value);
-#else
-  time_t time_value = static_cast<time_t>(value);
-  gmtime_r(&time_value, &date);
-#endif
+void GetTimeForSecondsSinceEpoch(std::tm& out_tm, int64_t seconds_since_epoch) {
+  try {
+    std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> timepoint{
+        std::chrono::seconds{seconds_since_epoch}};
+
+    std::chrono::year_month_day ymd = std::chrono::floor<std::chrono::days>(timepoint);
+
+    std::chrono::hh_mm_ss<std::chrono::seconds> timeofday{
+        timepoint - std::chrono::floor<std::chrono::days>(timepoint)};
+
+    std::memset(&out_tm, 0, sizeof(std::tm));
+
+    out_tm.tm_year = static_cast<int>(ymd.year()) - 1900;
+    out_tm.tm_mon = static_cast<unsigned>(ymd.month()) - 1;
+    out_tm.tm_mday = static_cast<unsigned>(ymd.day());
+    out_tm.tm_hour = static_cast<int>(timeofday.hours().count());
+    out_tm.tm_min = static_cast<int>(timeofday.minutes().count());
+    out_tm.tm_sec = static_cast<int>(timeofday.seconds().count());
+  } catch (const std::exception&) {
```

Comment:
@alinaliBQ I'm not sure we want to ignore exceptions here. Ideally this function would be `Result<std::tm> GetTimeForSecondsSinceEpoch(int64_t)` (or `Status ...` with an out parameter if there's a reason to do that. Also note that conventionally out parameters go at the end: https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs)

Reply:
However that is unrelated to the issue at hand so we can follow up in a new issue

Reply:
...Actually, why are we catching an exception in the first place? I don't think anything here throws?

Reply:
My bad. Its even ok in terms of odbc to throw error here